src/tests-rosetta/api.ts

```diff
@@ -1074,6 +1090,717 @@ describe('Rosetta API', () => {
+    const publicKey = publicKeyToString(pubKeyfromPrivKey(testnetKeys[0].secretKey));
+
+    const txOptions: UnsignedTokenTransferOptions = {
+      publicKey: publicKey,
+      recipient: standardPrincipalCV(testnetKeys[1].stacksAddress),
+      amount: new BigNum(12345),
+      network: GetStacksTestnetNetwork(),
+      memo: 'test memo',
+      nonce: new BigNum(0),
+      fee: new BigNum(200),
+    };
+
+    const unsignedTransaction = await makeUnsignedSTXTokenTransfer(txOptions);
+    const unsignedSerializedTx = unsignedTransaction.serialize().toString('hex');
+
+    const signer = new TransactionSigner(unsignedTransaction);
+    signer.signOrigin(createStacksPrivateKey(testnetKeys[0].secretKey));
+    const signedSerializedTx = unsignedTransaction.serialize().toString('hex');
+
+    const signature = getSignature(unsignedTransaction);
+    if (!signature) return;
+
+    const request: RosettaConstructionCombineRequest = {
+      network_identifier: {
+        blockchain: 'stacks',
+        network: 'testnet',
+      },
+      unsigned_transaction: unsignedSerializedTx,
+      signatures: [
+        {
+          signing_payload: {
+            hex_bytes: signature.data,
+            signature_type: 'ecdsa',
+          },
+          public_key: {
+            hex_bytes: publicKey,
+            curve_type: 'secp256k1',
+          },
+          signature_type: 'ecdsa',
+          hex_bytes: signature.data,
+        },
+      ],
+    };
+
+    const result = await supertest(api.server)
+      .post(`/rosetta/v1/construction/combine`)
+      .send(request);
+
+    expect(result.status).toBe(200);
+    expect(result.type).toBe('application/json');
+
+    const expectedResponse: RosettaConstructionCombineResponse = {
+      signed_transaction: signedSerializedTx,
+    };
+
+    expect(JSON.parse(result.text)).toEqual(expectedResponse);
+  });
+
+  test('combine multi sign success', async () => {
+    const publicKey1 = publicKeyToString(pubKeyfromPrivKey(testnetKeys[0].secretKey));
+    const publicKey2 = publicKeyToString(pubKeyfromPrivKey(testnetKeys[1].secretKey));
+
+    const txOptions: UnsignedMultiSigTokenTransferOptions = {
+      publicKeys: [publicKey1, publicKey2],
+      numSignatures: 2,
+      recipient: standardPrincipalCV(testnetKeys[1].stacksAddress),
+      amount: new BigNum(12345),
+      network: GetStacksTestnetNetwork(),
+      memo: 'test memo',
+      nonce: new BigNum(0),
+      fee: new BigNum(200),
+    };
+
+    const unsignedTransaction = await makeUnsignedSTXTokenTransfer(txOptions);
+    const unsignedSerializedTx = unsignedTransaction.serialize().toString('hex');
+
+    const signer = new TransactionSigner(unsignedTransaction);
+    signer.signOrigin(createStacksPrivateKey(testnetKeys[0].secretKey));
+    const signedSerializedTx = unsignedTransaction.serialize().toString('hex');
+
+    const signature = getSignature(unsignedTransaction);
+    if (!signature) return;
```

Comment:
The test exits here early without error

Reply:
I should fix this or remove this test and multi-sig support altogether depends on how we proceed on https://github.com/blockstack/stacks-blockchain-api/pull/275#discussion_r501574721.